nodetypes/radon.nodes.aws/AwsLambdaFunctionFromS3/NodeType.tosca

```diff
@@ -134,6 +136,22 @@ node_types:
             type: string
             required: false
             default: { get_property: [ SELF, env_vars ] }
+          s3_bucket_key:
+            type: string
+            required: true
+            default: { get_property: [ SELF, s3_bucket_key ] }
+          aws_region:
+            type: string
+            required: true
+            default: { get_property: [ SELF, host, region ] }
+          role_arn:
+            type: string
+            required: true
+            default: { get_attribute: [ SELF, host, role_arn ] }
+          function_name:
+            type: string
+            required: true
+            default: { get_property: [ SELF, name ] }
```

Comment:
related to the previous comment on `function_name`: here the property `name` is used - and it is inherited from the abstract Function type, what was the reason to introduce the `function_name` property?

Reply:
As i recall it, name property is inherited and thus it is not present in the nodetype definition of nodetypes/radon.nodes.aws/AwsLambdaFunctionFromS3/NodeType.tosca . I introduced funtion_name as a property in order to use its value in the monitoring functionality. Thus in the AWSIsMonitoredBy relationship definition (and in other places) function_name property is referenced.

Reply:
I think the changes a re ready. Can you please check..?

Reply:
thanks @sydkbc, `name` seems to be used correctly, have you tried deploying some existing template using this type after your changes?

Another question: was there a reason to delete the description for the `region` property?

Reply:
I did try to deploy and it actually worked so I assume it is ok. I also updated the relevant service templates that use these node types to reflect the update. As for the description consider it an error by my part.

Reply:
@sydkbc I just noticed that all inputs you added are already present:
-- `s3_bucket_key` is already defined on line 84
-- `aws_region` is already defined on line 88
-- `role_arn` is already defined on line 92
-- `function_name` is already defined on line 100

As a result, all changes introduced in this file are not needed - could you, please, restore this file to its original version and push this commit

Reply:
Done!